### ISSUE TITLE
extraneous input after successful compilation

### ISSUE THREAD

**Issue by hexpaydaydev** [sc-forks/solidity-coverage#799]
hi there, it seems like this may be happening on a separate layer from the compiler itself as hardhat does not seem to have a problem
line in question (key)
```
mapping(bytes32 key => uint256) public authorization;
```
build (compile) is successful
```
➜  stake-manager git:(master) ✗ yarn run build
yarn run v1.22.15
$ solhint 'contracts/**/*.sol' && hardhat compile

Generating typings for: 47 artifacts in dir: artifacts/types for target: ethers-v5
Successfully generated 108 typings!
Compiled 36 Solidity files successfully
✨  Done in 14.43s.
```
vs coverage
```
➜  stake-manager git:(master) ✗ npx hardhat coverage 

Version
=======
> solidity-coverage: v0.8.2

Instrumenting for coverage...
=============================

> AuthorizationManager.sol
Error in plugin solidity-coverage: Error: Could not instrument: AuthorizationManager.sol. (Please verify solc can compile this file without errors.) extraneous input 'key' expecting '=>' (13:18)

For more info run Hardhat with --show-stack-traces
```

**Comment by cgewecke** [comment]
Resolved in 0.8.4

**Comment by hexpaydaydev** [comment]
quick update, the errors appear to still be printing
```
Error found while parsing the following file: ./contracts/AuthorizationManager.sol

ParserError: extraneous input 'key' expecting '=>' (18:18)
    at Object.parse (~/hexpayday/stake-manager/node_modules/@solidity-parser/parser/dist/index.cjs.js:36973:11)
    at ~/hexpayday/stake-manager/node_modules/surya/lib/mdreport.js:60:23
    at _loop (~/hexpayday/stake-manager/node_modules/surya/lib/mdreport.js:69:6)
    at Object.mdreport (~/hexpayday/stake-manager/node_modules/surya/lib/mdreport.js:185:18)
    at SolidityMetricsContainer.generateReportMarkdown (~/hexpayday/stake-manager/node_modules/solidity-code-metrics/src/metrics/metrics.js:204:35)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  errors: [
    {
      message: "extraneous input 'key' expecting '=>'",
      line: 18,
      column: 18
    }
  ]
}
```
the process does not exit, which is great! but it would be great to not have these logs if possible

example repo can be found here: https://github.com/hexpay-day/stake-manager
